demo-android/src/main/java/nl/tudelft/ipv8/android/demo/ui/bitcoin/ImportKeyDialog.kt

```diff
@@ -0,0 +1,91 @@
+package nl.tudelft.ipv8.android.demo.ui.bitcoin
+
+import android.app.Dialog
+import android.os.Bundle
+import android.widget.Switch
+import android.widget.TextView
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import nl.tudelft.ipv8.android.demo.R
+
+class ImportKeyDialog : DialogFragment() {
+    private lateinit var listener: ImportKeyDialogListener
+
+    interface ImportKeyDialogListener {
+        fun onImport(address: String, privateKey: String, testNet: Boolean)
+        fun onImportDone()
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        try {
+            listener = targetFragment as ImportKeyDialogListener
+        } catch (e: ClassCastException) {
+            throw ClassCastException("Calling Fragment must implement Listener")
+        }
+    }
+
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        return activity?.let {
+            val builder = AlertDialog.Builder(it)
+            val inflater = requireActivity().layoutInflater
+            val view = inflater.inflate(R.layout.import_key_dialog, null)
+
+            builder.setView(view)
+                .setTitle("Import Existing Address")
+                .setPositiveButton(
+                    "Import"
+                ) { _, _ ->
+
+                    val pk = view.findViewById<TextView>(R.id.public_key_input)
+                    val sk = view.findViewById<TextView>(R.id.private_key_input)
+                    val netSwitch = view.findViewById<Switch>(R.id.net_switch)
+
+                    val address = pk.text.toString()
```

Comment:
You use the public key here to get the address of their wallet. However, if I recall correctly, the public key and the address are different. There is a method to get the address from the public key, I believe you should use this. I can't remember it from the top of my head, I'll look for it.

Reply:
That's not what's happening. Only the variable is called public key, but the value itself is an address. I'll change that.